website/server/libs/groupTasks.js

```diff
@@ -1,24 +1,73 @@
+import {model as Users} from '../models/user/index';
 
 const SHARED_COMPLETION = {
   default: 'recurringCompletion',
   single: 'singleCompletion',
   every: 'allAssignedCompletion',
 };
 
-async function _completeMasterTask (masterTask) {
-  masterTask.completed = true;
+async function _completeOrUncompleteMasterTask (masterTask, completed) {
+  masterTask.completed = completed;
   await masterTask.save();
 }
 
-async function _deleteUnfinishedTasks (groupMemberTask) {
-  await Tasks.Task.deleteMany({
-    'group.taskId': groupMemberTask.group.taskId,
-    $and: [
-      {userId: {$exists: true}},
-      {userId: {$ne: groupMemberTask.userId}},
-    ],
-  }).exec();
+async function _updateAssignedUsersTasks (masterTask, groupMemberTask) {
+  if (groupMemberTask.type == 'todo') {
+    if (groupMemberTask.completed) {
+      // The task was done by one person and is removed from others' lists
+      await Tasks.Task.deleteMany({
+        'group.taskId': groupMemberTask.group.taskId,
+        $and: [
+          {userId: {$exists: true}},
+          {userId: {$ne: groupMemberTask.userId}},
+        ],
+      }).exec();
+    } else {
+      // The task was uncompleted by the group member and should be recreated for assignedUsers
+      let group = await Groups.findById(masterTask.group.id);
+      let userList = [];
+      masterTask.group.assignedUsers.forEach(userId => {
+        let query = {_id: userId};
+        userList.push(query);
+      });
+      // @REVIEW There has to be a better way to do this
+      // async arrow function callback to the find and/or a forEach resulted in Mongoose ParallelSaveErrors
+      let assignedUsers = await Users.find({
+          $or: userList
+        }).exec();
+      for (let i=0; i < assignedUsers.length; i++) {
+        let promises = [];
+        promises.push(group.syncTask(masterTask, assignedUsers[i]));
+        promises.push(group.save());
+        await Promise.all(promises);
+      }
+    }
+  } else {
+    // Complete or uncomplete the task on other users' lists
+    await Tasks.Task.find({
+      'group.taskId': groupMemberTask.group.taskId,
+      $and: [
+        {userId: {$exists: true}},
+        {userId: {$ne: groupMemberTask.userId}}
+      ]},
+      function (err, tasks) {
+        // @REVIEW How does Habitica handle errors?
+        if (err) return;
+
+        tasks.forEach (task => {
+          // Ajdust the task's completion to match the groupMemberTask
+          // @REVIEW Completed or notDue tasks have no effect at cron
+          // This maintain's the user's streak without scoring the task if someone else completed the task
+          // If no assignedUser completes the due daily, all users lose their streaks at their cron
+          // An alternative is to set the other assignedUsers' tasks to a later startDate
+          // Should we break their streaks to encourage competition for the daily?
+          task.completed = groupMemberTask.completed;
```

Comment:
This is the main point of feature discussion.
I based my implimentation (complete other users' dailies) on cron: https://github.com/HabitRPG/habitica/blob/develop/website/server/libs/cron.js#L310

The implications of completing the task are that:
* The user who completes the Daily will score the task
* Other users will see the Daily is completed, will not score the task, will not lose their streak or take damage

More discussion on how the feature should work may be found in #10690